standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -677,10 +656,6 @@ public boolean isActiveTransaction() {
    */
   @Override
   public void rollbackTransaction() {
-    if (openTrasactionCalls < 1) {
```

Comment:
is this a main fix?

Reply:
yes, we need the rollback in case of `openTrasactionCalls` is 0.
`openTrasactionCalls` is to tell when the transaction should commit, we need to very take care of `openTrasactionCalls` as it lives through the ObjectStore

Reply:
Can we come up with any kind of unit test that reproduces the starvation or something else that demonstrates the necessity of this fix?

Reply:
we could create a spy for PersistenceManager and mock the currentTransaction to fail on commit

Reply:
This original can only fix the leak on primary pool, the secondary pool can still have such issue, https://github.com/dengzhhu653/hive/commit/7b4a66cf030e8570214341424bab258520abe919 can reproduce the problem